[MRG] Update to py 3.13 for doc build

DoomedJupiter opened the pull request:
Update the python version used to build the docs. Change made in accordance with [rtd config](https://docs.readthedocs.com/platform/stable/config-file/v2.html#build-tools-python). 

Supports PR #551

bosd commented:
Oops, Maybe I approved too quickly.
Docs-build and xdoctest still mentions python 3.10
Will look later into this, when more time.

DoomedJupiter commented:
Updated the py version used to run the docs-build test in `tests.yml`. Also updated `noxfile.py` to use py v3.13. I don't quite understand the syntax here as it looks like we are forcing the py version with `python_verion[5]` even though `tests.yml` appears to be pushing py v3.13. 

The test `xdoctest` looks to be some sort of test on docstrings using the [xdoctest](https://github.com/Erotemic/xdoctest) package. It seems like this is not related to the sphinx bump so I left it as is, using py v3.10
